src/azure-cli/azure/cli/command_modules/acr/check_health.py

```diff
@@ -337,8 +337,75 @@ def _check_registry_health(cmd, registry_name, ignore_errors):
             _handle_error(CMK_MANAGED_IDENTITY_ERROR.format_error_message(registry_name), ignore_errors)
 
 
+def _check_private_endpoint(cmd, registry_name, vnet_of_private_endpoint):
+    import socket
+    from msrestazure.tools import parse_resource_id, is_valid_resource_id, resource_id
+    from azure.cli.core.profiles import ResourceType
+    from azure.cli.core.commands.client_factory import get_mgmt_service_client
+
+    if registry_name is None:
+        logger.warning("Registry name must be provided to verify DNS routings of its private endpoints")
+        return
+
+    registry = None
+
+    # retrieve registry
+    try:
+        registry, _ = get_registry_by_name(cmd.cli_ctx, registry_name)
+    except CLIError:
+        logger.warning('Registry resource of "%s" must be accessible to verify DNS routings of its private endpoints.',
+                       registry_name)
+        return
+
+    if not registry.private_endpoint_connections:
+        logger.warning('Registry "%s" doesn\'t have private endpoints to verify DNS routings.', registry_name)
+        return
+
+    if not is_valid_resource_id(vnet_of_private_endpoint):
+        res = parse_resource_id(registry.id)
+        vnet_of_private_endpoint = resource_id(name=vnet_of_private_endpoint, resource_group=res['resource_group'],
```

Comment:
@yugangw-msft Do you think we should validate if the user has entered a valid VNET resource ID? Right now, we only check if it's a valid resource ID.

For instance, if a user (by mistake) provides a Registry ID instead of VNET ID, the current code flow will log an error:
>Registry ... doesn\t have private endpoints in the vnet of ...